tests/sqlparser_common.rs

```diff
@@ -8685,3 +8685,263 @@ fn parse_map_access_expr() {
         let _ = dialects.verified_expr(sql);
     }
 }
+
+#[test]
+fn test_match_recognize() {
+    let table = TableFactor::Table {
+        name: ObjectName(vec![Ident::new("my_table")]),
+        alias: None,
+        args: None,
+        with_hints: vec![],
+        version: None,
+        partitions: vec![],
+    };
+    fn check(options: &str, expect: TableFactor) {
+        let select = all_dialects_where(|d| d.supports_match_recognize()).verified_only_select(
+            &format!("SELECT * FROM my_table MATCH_RECOGNIZE({options})"),
+        );
+        assert_eq!(&select.from[0].relation, &expect);
+    }
+    check(
+        concat!(
+            "PARTITION BY company ",
```

Comment:
can we add some simpler roundtrip-only tests that parses statements without the optional parts of match recognize like `partition by`?